crates/sui/src/unit_tests/snapshots/sui__upgrade_compatibility__upgrade_compatibility_tests__struct.snap

```diff
@@ -48,46 +48,167 @@ error[Compatibility E01003]: ability mismatch
    = Restore the original abilities of struct 'AddMultipleAbilities': none.
 
 error[Compatibility E01002]: type mismatch
-   ┌─ /fixtures/upgrade_errors/struct_errors_v2/sources/UpgradeErrors.move:17:19
+   ┌─ /fixtures/upgrade_errors/struct_errors_v2/sources/UpgradeErrors.move:18:19
    │
-17 │     public struct AddField {
+18 │     public struct AddFieldToEmpty {
+   │                   ^^^^^^^^^^^^^^^ Incorrect number of fields: expected 0, found 1
```

Comment:
(thinking out loud) I was thinking that this error should show on the fields but maybe that's not the case.

Reply:
This is a pattern I'm following from the move compiler. better to tell the user a mismatch of number of fields first since a diff could be quite large if the user deletes the first argument/field/param etc and not very helpful.